This is quite similar in nature to:

https://github.com/ansible/awx/pull/3305

We have not committed to supporting inventory updates via scripts in python3 virtual environments. This shouldn't be a formal test objective.

We will _absolutely_ support inventory updates via plugins in python3 virtual environments, including the default virtual environment.

However, it remains extremely important that `tower.py` continues to work in python2.

```
$ python
Python 2.7.11 (default, Oct 17 2016, 14:59:40) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from urllib.parse import urljoin
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named parse
```

At best, you could do a conditional import. I wouldn't object to that, but it would still be officially unsupported to run it in python3.

--- a/awx/plugins/inventory/tower.py
+++ b/awx/plugins/inventory/tower.py
@@ -34,7 +34,11 @@ import sys
 import json
 import requests
 from requests.auth import HTTPBasicAuth
-from urlparse import urljoin
+
+try:
+    from urlparse import urljoin
+except ImportError:
+    from urllib.parse import urljoin
 
 
 def parse_configuration():

